File: AmsterdamEats.Maui/AmsterdamEats.Maui/Models/Restaurant.cs
using AmsterdamEats.Config;

namespace AmsterdamEats.Models;

public class Restaurant
{
    public string Id { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public string Address { get; set; } = string.Empty;
    public double Rating { get; set; }
    public int UserRatingCount { get; set; }
    public string? PriceLevelDisplay { get; set; }
    public string? WebsiteUri { get; set; }
    public string? PhotoName { get; set; }
    public int? ZenchefRestaurantId { get; set; }
    public List<ZenchefSlot> AvailableSlots { get; set; } = [];
    public bool AvailabilityChecked { get; set; }

    public bool HasAvailability => AvailableSlots.Count > 0;
    public bool IsBookable => ZenchefRestaurantId.HasValue;
    public bool HasWebsite => !string.IsNullOrEmpty(WebsiteUri);

    public string RatingDisplay => $"★ {Rating:F1}";
    public string ReviewCountDisplay => $"({UserRatingCount:N0} reviews)";
    public string AvailabilityBadge => HasAvailability ? "Available"
        : (AvailabilityChecked && !IsBookable ? "No booking" : string.Empty);
    public bool ShowAvailabilityBadge => !string.IsNullOrEmpty(AvailabilityBadge);

    public string PhotoUrl => PhotoName is not null
        ? $"{ApiConfig.GooglePlacesBaseUrl}/{PhotoName}/media?maxHeightPx=400&maxWidthPx=400&key={ApiConfig.GooglePlacesApiKey}"
        : string.Empty;
}

File: AmsterdamEats/AmsterdamEats/Models/Restaurant.swift
import Foundation

struct Restaurant: Identifiable {
    let id: String
    let name: String
    let address: String
    let rating: Double
    let userRatingCount: Int
    let priceLevel: PriceLevelDisplay?
    let websiteURI: String?
    let photoName: String?
    var zenchefRestaurantId: Int?
    var availableSlots: [ZenchefSlot] = []
    var availabilityChecked: Bool = false

    enum PriceLevelDisplay: String, Decodable {
        case unspecified = "PRICE_LEVEL_UNSPECIFIED"
        case free = "PRICE_LEVEL_FREE"
        case inexpensive = "PRICE_LEVEL_INEXPENSIVE"
        case moderate = "PRICE_LEVEL_MODERATE"
        case expensive = "PRICE_LEVEL_EXPENSIVE"
        case veryExpensive = "PRICE_LEVEL_VERY_EXPENSIVE"

        var displayString: String {
            switch self {
            case .unspecified, .free: return ""
            case .inexpensive: return "€"
            case .moderate: return "€€"
            case .expensive: return "€€€"
            case .veryExpensive: return "€€€€"
            }
        }
    }

    var photoURL: URL? {
        guard let photoName else { return nil }
        return URL(string: "\(APIConfig.googlePlacesBaseURL)/\(photoName)/media?maxHeightPx=400&maxWidthPx=400&key=\(APIConfig.googlePlacesAPIKey)")
    }

    var hasAvailability: Bool { !availableSlots.isEmpty }
    var isBookable: Bool { zenchefRestaurantId != nil }
}
